pkg/cloudscale/objectstorage.go

```diff
@@ -61,8 +68,29 @@ func (o *ObjectStorage) GetMetrics(ctx context.Context, billingDate time.Time) (
 	if err != nil {
 		o.providerMetrics["providerFailed"].Inc()
 		return nil, err
-	} else {
-		o.providerMetrics["providerSucceeded"].Inc()
+	}
+
+	bucketMap := make(map[string]*ObjectStorageData)
+
+	// create a map with bucket name as key, this way we match buckets created manually and not via Appcat service
+	for i, bucketMetric := range bucketMetrics.Data {
+		bucketMap[bucketMetric.Subject.BucketName] = &ObjectStorageData{
+			BucketMetricsData: bucketMetrics.Data[i],
+		}
+	}
+
+	// Since our buckets are always created in the convention $namespace.$bucketname, we can extract the namespace from the bucket name by splitting it.
+	// However, we need to fetch the user details to get the actual namespace.
+	for _, bucket := range bucketMap {
+		// fetch bucket user by id
+		logger.Info("fetching user details", "userID", bucket.Subject.ObjectsUserID)
+		userDetails, err := o.client.ObjectsUsers.Get(ctx, bucket.Subject.ObjectsUserID)
+		if err != nil {
```

Comment:
Trying here to follow the logic and wanted to double check with you. In case the bucket doesn't have a user object (bucket that are created manually) the code seems to return an error right away instead of just skipping it. Is that what is wanted?

Reply:
yes, because in Cloudscale we have separate namespace for Appcat Buckets/Objects - no other objects/users/buckets should live there

Reply:
yeah but then the whole metrics break when there's a new manual object in cloudscale, why not log it and move forward? I don't see why we have to return an error instead of just logging it